lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,310 @@
+				lineIndex += 1;
+			} else if ( completionIndex + 1 === completion.length ) {
+				lineIndex += 1;
+				completionIndex = lastMatchedIndex + 1;
+			}
+			completionIndex += 1;
+		}
+
+		// Highlight stored indexes in the completion string:
+		for ( i = 0; i < completion.length; i++ ) {
+			if ( boldIndexes.includes( i ) ) {
+				highlighted += '\x1b[1m' + completion[i] + '\x1b[0m';
+			} else {
+				highlighted += completion[ i ];
+			}
+		}
+		return highlighted;
+	});
+});
+
+/**
+* Displays the completions to the output stream.
+*
+* @private
+* @name _displayCompletions
+* @memberof CompleterEngine.prototype
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_displayCompletions', function displayCompletions() {
+	var highlightedCompletions;
+	var completionsLength;
+	var widthOfColumn;
+	var whitespaces;
+	var completion;
+	var lineIndex;
+	var columns;
+	var output;
+	var i;
+
+	// Determine number of columns of completions that should be displayed to the output stream
+	completionsLength = this._completionsList.map( getLength );
+	widthOfColumn = max( completionsLength.length, completionsLength, 1 ) + 4; // 4 space padding
+	columns = floor( this._ostream.columns / widthOfColumn ) || 1;
+
+	// Highlight completions if operating in "terminal" mode...
+	if ( this.repl._isTTY ) {
+		highlightedCompletions = this._highlightCompletions();
+	} else {
+		highlightedCompletions = this._completionsList;
+	}
+
+	output = '\r\n';
+	lineIndex = 0;
+	whitespaces = 0;
+	for ( i = 0; i < highlightedCompletions.length; i++ ) {
+		completion = highlightedCompletions[ i ];
+		if ( lineIndex === columns ) {
+			// Reached end of column, enter next line:
+			output += '\r\n';
+			lineIndex = 0;
+			whitespaces = 0;
+		} else {
+			// Fill the space to move to the next column:
+			output += repeat( ' ', whitespaces );
+		}
+		// Add completion string to the column in output:
+		output += completion;
+		whitespaces = widthOfColumn - completionsLength[i];
+		lineIndex += 1;
+	}
+	// Ensure a space between completions and the following prompt...
+	if ( lineIndex !== 0 ) {
+		output += '\r\n\r\n';
+	}
+	// Write completions to the output stream:
+	this._ostream.write( output );
+});
+
+/**
+* Callback which should be invoked **before** a "keypress" event is processed by a readline interface.
+*
+* @name beforeKeypress
+* @memberof CompleterEngine.prototype
+* @param {string} data - input data
+* @param {Object} key - key object
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, 'beforeKeypress', function beforeKeypress() {
+	var cursor;
+	var line;
+	var self = this;
+
+	cursor = this._rli.cursor;
+	line = this._rli.line;
+
+	// Get the line before the cursor:
+	this._line = line.slice( 0, cursor );
+
+	// Pause the input stream before generating completions as the completer may be asynchronous...
+	this._rli.pause();
+	this._completer( this._line, function onCompletions( error, completions ) {
```

Comment:
This is restricted syntax. The only time we inline function declarations is when defining prototype properties (e.g., L288.